src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new ArrayList[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        ensureCapacity();
+        int index = getIndexFromKey(hash(key), buckets.length);
+        List<Entry<K, V>> bucket = buckets[index];
+        if (bucket == null) {
+            bucket = new ArrayList<>();
+            buckets[index] = bucket;
+        }
+        for (Entry<K, V> entry : bucket) {
+            if (key == null && entry.key == null
+                    || key != null && key.equals(entry.key)) {
```

Comment:
extract to a private method or use Objects.equals, from this HW on we can use it